src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,90 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { Loader } from '../Loader';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { User } from '../../types/User';
+import { useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const todo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [hasReceivedUser, setHasReceivedUser] = useState(false);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        if (todo) {
+          const fetchedUser = await getUser(todo.userId);
+
+          setUser(fetchedUser);
+          setHasReceivedUser(true);
+        }
+      } catch {
+        setHasError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchUser();
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading
+        ? <Loader />
+        : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${todo?.id}`}
+              </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => dispatch(currentTodoActions.removeTodo())}
+              />
+            </header>
+
+            <div className="modal-card-body">
+              {hasError && (
+                <strong className="has-text-success">
+                  Unable to load the user
+                </strong>
+              )}
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+              {hasReceivedUser && (
```

Comment:
It could be done this way:
```suggestion
              {user && (
```
Try to avoid creating additional states when you can handle the required functionality without it.